eclipse/src/saros/ui/util/CollaborationUtils.java

```diff
@@ -58,44 +56,37 @@ private CollaborationUtils() {
   }
 
   /**
-   * Starts a new session and shares the given resources with given contacts.<br>
-   * Does nothing if a {@link ISarosSession session} is already running.
+   * Starts a new session and shares the given reference points with the given contacts.
    *
-   * @param resources
-   * @param contacts
-   * @nonBlocking
-   */
-  public static void startSession(final List<IResource> resources, final List<JID> contacts) {
-    // TODO remove assertion that only projects are shared (resolved in followup PR)
-    assert resources.stream().allMatch(resource -> resource instanceof IProject)
-        : "Encountered non-project resource to share";
-
-    final Set<IProject> projects =
-        resources.stream().map(resource -> (IProject) resource).collect(Collectors.toSet());
-
-    startSession(projects, contacts);
-  }
-
-  /**
-   * Starts a new session and shares the given projects with given contacts.<br>
-   * Does nothing if a {@link ISarosSession session} is already running.
+   * <p>Does nothing if a {@link ISarosSession session} is already running.
    *
-   * @param projects the projects share
-   * @param contacts the contacts to share the projects with
+   * @param referencePoints the reference points to share
+   * @param contacts the contacts to share the reference points with
    * @nonBlocking
    */
-  public static void startSession(final Set<IProject> projects, final List<JID> contacts) {
+  public static void startSession(
+      final Set<IReferencePoint> referencePoints, final List<JID> contacts) {
```

Comment:
As this is mainly (only?) called by Eclipse what is the problem to chance this into a set of IContainer instead ?

Reply:
I see your point. It would avoid the to unwrap the object to access the delegate. I will change it.

I implemented it this way to match the method in `ICollaborationUtils`. But since I removed that interface and its implementations (as it was only used by the HTML UI way back), that is no longer applicable.

Reply:
Done.